xapi/xapi.go

```diff
@@ -92,6 +115,7 @@ func (r *selectResult) SetFields(fields []*types.FieldType) {
 
 // Close closes SelectResult.
 func (r *selectResult) Close() error {
+	close(r.done)
 	return r.resp.Close()
```

Comment:
we'd better add a `r.results = nil`  here.
when error happened, r.results is still refered by selectResult, then the data in the channel won't be freed by GC.